google-cloud-storage/src/main/java/com/google/cloud/storage/StorageImpl.java

```diff
@@ -1279,6 +1281,40 @@ public Boolean call() {
     }
   }
 
+  @Override
+  public Map<BucketInfo.LifecycleRule, Boolean> deleteLifecycleRules(
+      String bucket, BucketInfo.LifecycleRule... rules) {
+    final Storage storage = getOptions().getService();
+    final Map<BucketInfo.LifecycleRule, Boolean> results = Maps.newHashMap();
+    try {
+      Bucket remoteBucket =
+          storage.get(bucket, Storage.BucketGetOption.fields(Storage.BucketField.LIFECYCLE));
+      ArrayList<BucketInfo.LifecycleRule> lifecycleRules =
+          new ArrayList(remoteBucket.getLifecycleRules());
+      Iterator<BucketInfo.LifecycleRule> ruleIterator = lifecycleRules.iterator();
+      for (BucketInfo.LifecycleRule rule : rules) {
+        while (ruleIterator.hasNext()) {
+          BucketInfo.LifecycleRule actualRule = ruleIterator.next();
+          if (rule.getAction().getActionType().equals(actualRule.getAction().getActionType())) {
+            ruleIterator.remove();
+            results.put(rule, Boolean.TRUE);
+          }
+        }
+      }
+      if (!results.isEmpty()) {
+        storage
+            .get(bucket, Storage.BucketGetOption.fields())
+            .toBuilder()
+            .setLifecycleRules(lifecycleRules)
+            .build()
+            .update();
+      }
+      return Collections.unmodifiableMap(results);
+    } catch (StorageException ex) {
+      throw ex;
+    }
+  }
```

Comment:
The returned type is `Map<BucketInfo.LifecycleRule, Boolean>`
But there returned object will never contain 'rule -> false', only 'rule -> true'
Is it really necessary to return a map? Will it be enough to return a list?

Reply:
the purpose of `map<BucketInfo.LifecycleRule, Boolean>` is to confirm rules deleted successfully and am agreed with this `returned object will never contain rule -> false, only rule -> true.`  it's returning either `'rule -> true'` or `empty `if rule not found.
>Will it be enough to return a list?
you mean list of `deleted rule` or list of `boolean`?

Reply:
Not list of boolean. You can return either list of deleted rules or the list of rules, which were not deleted. If you want to return a map rule->boolean, you need to initialize one with rule->false.
Then for existing rules you will update rule->true.

Reply:
done, planning to go with list of deleted rules